### ISSUE TITLE
[1.17+] Intel GPU HD 4000 crashes with shaders

### ISSUE THREAD

**Issue by foulmist** [sp614x/optifine#6072]
## Installation Method
Installed using installer

## Other Installed Mods
none

shaders work fine with 1.16.5 Optifine

**Comment by fayer3** [comment]
do you use any resource pack? does it also happen without a resourcepack?

**Comment by foulmist** [comment]
I tried all the available settings and scenarios. With and without resource packs, with fast render on/off, basically everything you can think of. Even reinstalled previous versions of the Intel Graphics drivers on Win7,10. Tried with Linux but couldn't get it to start because it didn't properly recognize the openGL version for some reason even though I installed new kernels and mesa drivers. (1st time with Linux so I might have done something wrong).

tried different java versions too, and different launchers (original and TLauncher) 
tried all the Optifine versions for 1.17 and 1.17.1 - nothing works EXCEPT for the Internal Shaders! They work fine.

But when I try to load any other shader it just crashes with that exit code mention in the OP.
Even tried 2 Chunks and Render and Shadow Quality down to 0.5x
Basically I tried all the LOWEST Settings and mid ones.

No problems on 1.16.5 though.

**Comment by fayer3** [comment]
can you post a F3 screenshot without shaders, since they crash the game?
They work fine on my HD 4400
![2021-07-24_13 18 04](https://user-images.githubusercontent.com/8282299/126867617-65876e22-a53b-4fda-9e94-2891051a7285.png)

**Comment by foulmist** [comment]
HD 4400 works fine I have tried it on another PC. The problem is on HD 4000 only! 3rd Gen Ivy Bridge GPU! on Intel 3rd Gen CPUS like i3 3310M or i5 3320M or i7 i3630QM, etc.
